project_template/inc/az/exampleshortname/operations.h

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/** @file operations.h */
+#pragma once

```

Comment:
not all compiler support #pragma once. However, all compilers will support 
#ifndef OPERATIONS_H
#define OPERATIONS_H..


#endif

Reply:
pragma once is supported by all modern compilers except for Cray Classic and TASKING VX. If we want to support those compilers then it is very, very easy to switch from #pragma once to include guards using a little bit of regex.

Also `#pragma once` is much better in the case where you copied / moved / symlinked / somehow got a file from more than one place, since it will likely give you warnings / errors instead of silently compiling.

Reply:
`#pragma once` is awesome feature but I thought we should limit ourselves to C99.

Reply:
Did you try that on Arduino compilers, including the ESP8266 and ESP32? MCU manufactures didn't claim that their compilers are fully C99 and they may not have these kind of `pragma`.

Reply:
Those seem to use a gcc based toolchain which would support this pragma